samtranslator/swagger/swagger.py

```diff
@@ -898,31 +900,38 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
             resource_list = self._get_method_path_uri_list(path, api_id, stage)
             self._add_ip_resource_policy_for_method(ip_range_blacklist, "IpAddress", resource_list)
 
-        if (
-            (source_vpc_blacklist is not None)
-            or (source_vpc_intrinsic_blacklist is not None)
-            or (source_vpce_intrinsic_blacklist is not None)
-        ):
-            blacklist_dict = {
-                "StringEndpointList": source_vpc_blacklist,
-                "IntrinsicVpcList": source_vpc_intrinsic_blacklist,
-                "IntrinsicVpceList": source_vpce_intrinsic_blacklist,
-            }
-            resource_list = self._get_method_path_uri_list(path, api_id, stage)
-            self._add_vpc_resource_policy_for_method(blacklist_dict, "StringEquals", resource_list)
+        if source_vpc_blacklist is not None and not all(isinstance(x, string_types) for x in source_vpc_blacklist):
```

Comment:
Do we have any generic methods to check list types? I assume this could be a fairly common operation.

Reply:
In fact just noticed it's repeated underneath. Can we refactor that into its own function?

Reply:
@hoffa What's the proposal here? I don't really understand the need to make this into it's own function. To me that function will just return a bool and keep the if, so we can keep the context in one place. From that, it really doesn't make sense to me to drop it into a method but could be missing an alternative solution.

Ideally, this should be all in one place (the spec). Since we don't or that it's a much bigger change. I am proposing to keep this as is and stop the failures/crashing on SAM. We can later build this into the spec to solve it at the root.

Reply:
My point is that `if source_vpc_blacklist is not None and not all(isinstance(x, string_types) for x in source_vpc_blacklist)` is verbose and adds cognitive load to understand for what's a simple and generic check ("are the values in `source_vpc_blacklist` strings?").

My proposal is to have something like:

```python
def is_iterable_type(xs: Optional[Iterable], types: Tuple[Type]):
    if not xs:
        return False
    return all(isinstance(x, types) for x in xs)
```

And then replace those checks by:

```python
if not is_iterable_type(source_vpc_blacklist, string_types):
```

Reply:
Could we maybe reuse `list_of` from `samtranslator/model/types.py`?

Reply:
@hoffa Fair enough.

@mgrandis We can but honestly not sure if that is the right thing. The `types.py` is used as part of the Spec validation. Ideally, these things should be caught before we get into the generation of swagger, but we aren't doing validation on the Values for these properties (no idea why that is). If we use `types.py` here, my concern is that we start blending the validation logic. It could be the right thing, just not sure if it is. I know you have lots of thoughts on the validation parts. Do you have concerns in mixing this? 

It would look something like:
```
list_validator = list_of(is_str())
if not list_validator(source_vpc_blacklist, should_raise=False):
    raise
```